sdk/internal/recording/recording.go

```diff
@@ -438,7 +439,37 @@ var modeMap = map[RecordMode]recorder.Mode{
 	Playback: recorder.ModeReplaying,
 }
 
-var recordMode = os.Getenv("AZURE_RECORD_MODE")
+func init() {
+	recordMode = os.Getenv("AZURE_RECORD_MODE")
+	if !(recordMode == RecordingMode || recordMode == PlaybackMode || recordMode == LiveMode) {
+		log.Printf("AZURE_RECORD_MODE was not understood, options are %s, %s, or %s Received: %v. Defaulting to playback", RecordingMode, PlaybackMode, LiveMode, recordMode)
+		recordMode = PlaybackMode
+	}
+
+	localFile, err := findProxyCertLocation()
+	if err != nil {
+		log.Println("Could not find the PROXY_CERT environment variable and was unable to locate the path in eng/common")
+	}
+
+	rootCAs, err = x509.SystemCertPool()
+	if err != nil && strings.Contains(err.Error(), "system root pool is not available on Windows") {
+		rootCAs = x509.NewCertPool()
+	} else if err != nil {
+		log.Println("could not create a system cert pool")
+	}
+
+	cert, err := ioutil.ReadFile(localFile)
+	if err != nil {
+		log.Printf("could not read file set in PROXY_CERT variable at %s.\n", localFile)
+	}
+
+	if ok := rootCAs.AppendCertsFromPEM(cert); !ok {
+		log.Println("no certs appended, using system certs only")
```

Comment:
What happens when `cert` is already in the pool?

Reply:
I can't find anything on duplicated certificates so I'm not sure what to assume here. I would assume having the same certificate twice is fine?

Reply:
That's how I think it *should* work but don't know how it actually works, so I was asking out of curiosity. I think it's fine to start with this and adjust if necessary.